### PR TITLE
[fix bug 1190614] Build alternate, optimized /firefox/new/.

### PR DESCRIPTION
**Pull request by jpetto**
After review/merge is complete, will test with Optimizely. See [bug 1197271](https://bugzilla.mozilla.org/show_bug.cgi?id=1197271).

### CODE REVIEW

**Comment by jpetto** [comment]
Branch is up on demo3:

https://www-demo3.allizom.org/en-US/firefox/new/2/

@alexgibson r?

File: bedrock/firefox/templates/firefox/new2.html [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+{% endblock %}
+
+{# all CSS must be in extrahead block for old IE #}
+{% block site_css %}{% endblock %}
+
+{% block extrahead %}
+  {% stylesheet 'firefox_new2' %}
+{% endblock %}
+
+{% block optimizely %}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/new2.html]: Is Optimizely needed on both the control page and this one?

File: media/css/firefox/new2.less [hunk 2]
```diff
@@ -0,0 +1,583 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+@import "../sandstone/lib.less";
+
+.sky #outer-wrapper {
+    background: #eaeff2;
+    background: -moz-radial-gradient(center, ellipse cover, rgba(234,239,242,1) 0%, rgba(212,221,228,.5) 60%),
```
Comment (alexgibson) [inline media/css/firefox/new2.less]: We can save a few bytes by removing all this prefix cruft. Probably just need the -webkit prefix?

File: media/js/firefox/new2.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+                    if (navigator.appVersion.indexOf('MSIE') !== -1) {
+                        return;
+                    }
+
+                    $directDownloadLink.trigger('click');
+                }
+            }
+
+            // prep both scenes for display
+            $downloadStarted.removeClass('invisible');
```
Comment (alexgibson) [inline media/js/firefox/new2.js]: Is there a reason why we need to rely on opacity here as opposed to using something like `visually-hidden`?

File: bedrock/settings/static_media.py [hunk 4]
```diff
@@ -1171,6 +1180,20 @@
...
     },
+    'firefox_new2': {
+        'source_filenames': (
+            'js/libs/jquery-1.11.3.min.js',
+            'js/libs/spin.min.js',
+            'js/base/global.js',
+            'js/base/global-init.js',
+            'js/base/search-params.js',
+            'js/newsletter/form.js',
+            'js/base/mozilla-input-placeholder.js',
```
Comment (alexgibson) [inline bedrock/settings/static_media.py]: Can we safely omit this file? I think only up-to-date Firefox users see the newsletter, in which case this polyfill shouldn't be needed.

File: media/js/firefox/new2.js [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+            // An iframe can not be used here to trigger the download because
+            // it will be blocked by Chrome if the download link redirects
+            // to a HTTP URI and we are on HTTPS.
+            function trackAndRedirect(url, virtualUrl) {
+                // Delay to initiate download is required to allow animation
+                // to finish loading in IE. If delay is removed, the DOM will
+                // unload before the animation completes and the page will
+                // stop in a half-animated state.
+                window.setTimeout(function() {
```
Comment (alexgibson) [inline media/js/firefox/new2.js]: Is this 500ms delay still needed now we removed the animation?

File: bedrock/firefox/templates/firefox/new2.html (hunk 1 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/new2.html]: Yep. :unamused:

File: media/js/firefox/new2.js (hunk 3 above)
Comment (jpetto) [inline media/js/firefox/new2.js]: Using `opacity` for the transition. JS decides which "scene" to show when the page loads, so thought a short transition from `opacity: 0;` to `opacity: 1;` was more pleasing. Any reason `visually-hidden` mixin would be preferable?

File: bedrock/settings/static_media.py (hunk 4 above)
Comment (jpetto) [inline bedrock/settings/static_media.py]: Ah, good catch! Removing.

File: media/js/firefox/new2.js (hunk 3 above)
Comment (alexgibson) [inline media/js/firefox/new2.js]: > Any reason visually-hidden mixin would be preferable?

Ah I see. I wasn't quite clear on why this bit was needed. I didn't really notice the transition effect when I tested the page locally, but I see it now. I had assumed you we're wanting to hide content until the page had loaded, which is why I suggested using `visibility` as this has broader support in old browsers. But if this is only for visual effect then we're probably all good?

File: media/css/firefox/new2.less (hunk 2 above)
Comment (jpetto) [inline media/css/firefox/new2.less]: I'm in with that. Should we keep both `-webkit` prefixes, or just the most recent?

File: media/js/firefox/new2.js (hunk 3 above)
Comment (jpetto) [inline media/js/firefox/new2.js]: Yeah, just a little visual nicety.

File: media/js/firefox/new2.js (hunk 5 above)
Comment (jpetto) [inline media/js/firefox/new2.js]: I'm going to remove it and put the branch on a demo to test in IE. Hopefully you're right. :+1:

File: media/css/firefox/new2.less [hunk 6]
```diff
@@ -0,0 +1,579 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+@import "../sandstone/lib.less";
+
+.sky #outer-wrapper {
+    background: #eaeff2;
+    background: -webkit-gradient(radial, center center, 0px, center center, 60%, color-stop(0%,rgba(234,239,242,.9)), color-stop(60%,rgba(212,221,228,.5))),
```
Comment (alexgibson) [inline media/css/firefox/new2.less]: I think we can get rid of this one too? Only the most recent -webkit prefixed version of the standard property should be needed still.

File: media/js/firefox/new2.js (hunk 3 above)
Comment (alexgibson) [inline media/js/firefox/new2.js]: One thought that just occurred to me - could this possibly effect download conversions negatively on slower connections? (e.g. waiting for the window load event to see the content?)

Comment (jpetto) [inline media/js/firefox/new2.js]: Hm, that is a good thought. They would see the headline and logo and such, but not the main content. I could make scene 1 visible by default, then fade in scene 2 on `window.load` if instructed by URL params.

This catch is that it's possible the download button would be visible but then disappear as the user is reaching for it. However, the download would start automatically in this case (on IE 9+, at least). Probably worth it?

**Comment by jpetto** [comment]
Tested removing the 500ms delay on the `trackAndRedirect` on [demo3](https://www-demo3.allizom.org/en-US/firefox/new/2/) in IE 7 & 8 and all seems well. :raised_hands:

**Comment by jpetto** [comment]
Went with the heavier, but seemingly better in this case, jQuery `fade` rather than transitioning opacity. Flow is much closer to original (scene 1 always loads, then transitions to scene 2), rather than loading scene 2 directly (when URL dictates) as in previous commit. Result is faster display of content (including download button) on slower internet connections.

Left this change in a separate commit in case we want to back it out.

[demo3](https://www-demo3.allizom.org/en-US/firefox/new/2/) has been updated with the latest.

**Comment by alexgibson** [comment]
Looks like Safari barfs a bit on the animation now that we removed the 500ms delay on the download :crying_cat_face:

File: media/js/firefox/new2.js [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+        if (!noScene2 && $('.download-button-wrapper:visible').length > 0) {
+            var $downloadButtonLinks = $('.download-button-wrapper .download-other-desktop').detach();
+            $downloadButtonLinks.css('display', 'block').insertBefore('#firefox-screenshot');
+        }
+
+        if (site.platform === 'android') {
+            $('#download-button-android .download-subtitle').html(
+                $('.android.download-button-wrapper').data('upgradeSubtitle'));
+
+            $downloadStarted.removeClass('invisible');
```
Comment (alexgibson) [inline media/js/firefox/new2.js]: Is this line and the one below still needed?

Comment (jpetto) [inline media/js/firefox/new2.js]: D'oh! Nope, not needed. Deleting.

**Comment by alexgibson** [comment]
Code looks good :+1: 

Tested demo3 on IE6-11, Edge, Chrome, Safari, Firefox, Opera and seems to all be working as expected.

r+ care to squarsh?

**Comment by jpetto** [comment]
Waiting until Optimizely test is ready before merge.

**Comment by jpetto** [comment]
As having the new URL live wont hurt anything, we're [good to merge before the Optimizely test is ready](https://bugzilla.mozilla.org/show_bug.cgi?id=1197271#c5).

**Pull request merged**